migrations/upgrade-1-8-0.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * Migration to create Smaily widget settings.
+ */
+
+$upgrade = function() {
+
+	$widgets = get_option( 'widget_smaily_widget', array() ); // Array sets empty default value.
+	foreach ( $widgets as &$widget ) {
+		if ( ! is_array( $widget ) ) {
+			continue;
+		}
+		$array2 = array(
```

Comment:
Use a variable name that conveys the meaning of the assigned value. For example `$new_settings` would work nicely.